proposed/2022/dotnet-nuget-why-proposal.md

```diff
@@ -0,0 +1,140 @@
+# dotnet nuget why command
+
+- Status: **Draft**
+- Author: [Kartheek Penagamuri](https://github.com/kartheekp-ms)
+- GitHub Issue [11782](https://github.com/NuGet/Home/issues/11782)
+
+## Problem Background
+
+Developers are often confused with top-level & transitive dependencies in their project. A developer should be able understand where every package originated from. A user should be able to promote a transitive dependency quickly & easily to a top-level dependency in the case there is a security concern on the package version depended on or if promoting to a top-level package will resolve a conflict.
+
+Knowing how a package is being included in a project helps a developer diagnose many problems. It also provides them with an opportunity to understand complex dependency chains they may have within their projects. There is not a great solution that exists to understand the nature of top-level packages and their transitive dependencies. The solution explorer in Visual Studio does the best job at allowing a user to dive into each NuGet package & see all transitive dependencies from the top-level dependency.
+
+![](../../meta/resources/TransitiveDependencies/SolutionView.png)
+
+The dotnet CLI does not provide the insight into “why” a transitive dependency is listed, although it does list everything that has been resolved.
+
+![](../../meta/resources/TransitiveDependencies/DotNetCLI.png)
+
+We could have considered a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of where the package came from. This could be a new column next to “Resolved” which says “Transitively Referenced” or “Referenced” and has a list of the top-level packages that requested the dependency. Tracking issue can be found here: https://github.com/NuGet/Home/issues/11625. Unfortunately, this solution doesn't provide an easy way to identify the dependency graph `(starting from project -> top-level package -> transitive-dependency (1) -> ..-> transitive dependency (n))` for a particular package.
```

Comment:
This sohuld be in the rationale and alternatives, not in the problem background.